sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -369,7 +370,11 @@ abstract class StreamExecution(
         //                      exception
         // UncheckedExecutionException - thrown by codes that cannot throw a checked
         //                               ExecutionException, such as BiFunction.apply
-        case e2 @ (_: UncheckedIOException | _: ExecutionException | _: UncheckedExecutionException)
+        // SparkException - thrown if the interrupt happens in the middle of an RPC wait
+        case e2 @ (_: UncheckedIOException |
+                   _: ExecutionException |
+                   _: UncheckedExecutionException |
+                   _: SparkException)
```

Comment:
I am not sure especially about this. It can be thrown for a great number of cases which aren't necessarily caused by the stop operation. I don't think it is a good idea to add it.

Reply:
I agree with the worry, but I'm not sure I see a better solution.

Other alternatives I can think of are matching against the specific exception message string, or changing ThreadUtils.awaitResult() to throw a custom exception. Do you have any thoughts?

Reply:
SGTM. I'll close this then.